cedar-policy/src/ffi/utils.rs

```diff
@@ -259,84 +259,234 @@ impl From<serde_json::Value> for Entities {
     }
 }
 
-#[derive(Debug, Serialize, Deserialize)]
+/// Represents a static policy in either the Cedar or JSON policy format
+#[derive(Debug, Serialize, Deserialize, Eq, PartialEq)]
+#[cfg_attr(feature = "wasm", derive(tsify::Tsify))]
+#[cfg_attr(feature = "wasm", tsify(into_wasm_abi, from_wasm_abi))]
 #[serde(untagged)]
 #[serde(
-    expecting = "policies as a concatenated string or multiple policies as a hashmap where the policy id is the key"
+    expecting = "expected a static policy in the Cedar or JSON policy format (with no duplicate keys)"
 )]
+pub enum Policy {
+    /// Policy in the Cedar policy format. See <https://docs.cedarpolicy.com/policies/syntax-policy.html>
+    Human(String),
+    /// Policy in Cedar's JSON policy format. See <https://docs.cedarpolicy.com/policies/json-format.html>
+    Json(JsonValueWithNoDuplicateKeys),
+}
+
+impl Policy {
+    /// Parse a [`Policy`] into a [`crate::Policy`]. Takes an optional id
+    /// argument that sets the policy id. If the argument is `None` then a
+    /// default id will be assigned.
+    pub(super) fn parse(self, id: Option<PolicyId>) -> Result<crate::Policy, miette::Report> {
+        let msg = id
+            .clone()
+            .map_or(String::new(), |id| format!(" with id `{id}`"));
+        match self {
+            Self::Human(str) => crate::Policy::parse(id.map(|id| id.to_string()), str)
```

Comment:
I wonder if we should have a `crate::Policy` constructor that takes a `PolicyId` instead of a `String` here, to avoid unparsing and reparsing.  Doesn't have to be in this PR.

Reply:
Made an issue: #1049